.github/workflows/pr.yml

```diff
@@ -0,0 +1,25 @@
+name: test
+on:
+  pull_request:
+    branches:
+      - master
+
+permissions: read-all
+
+jobs:
+  test:
+    name: 'Unit Test'
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Set up Go 1.20
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
```

Comment:
why shouldn't we use `actions/setup-go@v4.0.1` ?

Reply:
Best practice is to always pin to the sha digest because tags are mutable

Reply:
if anyone out there is listening "don't use tags immutably!!!!!"